From dfade66878f0ce833c140d02b5bcf6755b72415e Mon Sep 17 00:00:00 2001
From: Harry Youd <harry@harryyoud.co.uk>
Date: Sat, 5 Aug 2017 09:18:56 +0100
Subject: [PATCH 34/39] Fix extract_utils when not using blob sections

"$3" evaluates as "" when $3 is empty, and so $# is still 3
When $3 is empty, do not use it

Change-Id: I87df7dd8ed23cac3d4cff16dce0996bbe0a3e6cc
---
 build/tools/extract_utils.sh | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/build/tools/extract_utils.sh b/build/tools/extract_utils.sh
index 0e6ffc3b..b7350734 100644
--- a/build/tools/extract_utils.sh
+++ b/build/tools/extract_utils.sh
@@ -796,7 +796,11 @@ function extract() {
         exit 1
     fi
 
-    parse_file_list "$1" "$3"
+    if [ -z "$3" ]; then
+        parse_file_list "$1"
+    else
+        parse_file_list "$1" "$3"
+    fi
 
     # Allow failing, so we can try $DEST and/or $FILE
     set +e
-- 
2.15.0

